File: server/pbl03api/src/main/java/com/benkyousuru/pbl03api/model/service/ICustomerService.java
package com.benkyousuru.pbl03api.model.service;

import com.benkyousuru.pbl03api.model.model.CustomerModel;
import com.benkyousuru.pbl03api.model.model.LoginRequest;
import com.benkyousuru.pbl03api.model.model.LoginResponse;

public interface ICustomerService extends ICrudService<CustomerModel, Integer> {
    void setPassword(Integer customerId, String password);
    LoginResponse login(LoginRequest request);
    void logout(String token);
}

File: vite-client/src/interface/ICustomerService.ts
import { Customer } from "../models/Customer";
import { LoginRequest } from "../models/LoginRequest";
import { LoginResponse } from "../models/LoginResponse";

export interface ICustomerService {
  register(customer: Customer, password: string): Promise<void>;
  login(request: LoginRequest): Promise<LoginResponse>;
  logout(): void;
  update(): Promise<void>;
  changePassword(password: string): Promise<void>;

  loggedInCustomer: Customer | undefined;
}
